Bricks/net/tcp/impl/posix.h

```diff
@@ -509,20 +509,22 @@ struct Deleter {
 using addrinfo_t = std::unique_ptr<struct addrinfo, addrinfo_t_impl::Deleter>;
 
 inline addrinfo_t GetAddrInfo(const std::string& host, const std::string& serv = "") {
+  const std::string host_and_service = host + (serv.empty() ? "" : " " + serv);
   struct addrinfo* result = nullptr;
   struct addrinfo hints;
   memset(&hints, 0, sizeof(hints));
   hints.ai_family = AF_INET;
   hints.ai_socktype = SOCK_STREAM;
   hints.ai_protocol = IPPROTO_TCP;
   const int retval = ::getaddrinfo(host.c_str(), serv.c_str(), &hints, &result);
+  addrinfo_t safe_result(result);
   if (!result) {
-    CURRENT_THROW(SocketResolveAddressException(host + (serv.empty() ? "" : " " + serv)));
+    CURRENT_THROW(SocketResolveAddressException(host_and_service));
   } else if (retval) {
     freeaddrinfo(result);
-    CURRENT_THROW(SocketResolveAddressException(host + (serv.empty() ? "" : " " + serv) + " " + gai_strerror(retval)));
+    CURRENT_THROW(SocketResolveAddressException(host_and_service + gai_strerror(retval)));
```

Comment:
The space between the `host_and_service` and the error message from `gai_strerror` was lost here, so the error message will be glued with the `host_and_service`, example: `localhost:8080Some error message.` instead of `localhost:8080 Some error message.`